ref_builder/plan.py

```diff
@@ -28,34 +42,70 @@ def __str__(self) -> str:
         return f"{self.prefix} {self.key}"
 
 
-class Segment(BaseModel):
-    """The metadata of the segment."""
+class SequencePlan(BaseModel):
+    """Metadata and expected properties for an included sequence."""
 
     id: UUID4
-    """The unique id number of this segment"""
+    """The unique id number of the sequence plan"""
 
-    name: str
+    length: int
+    """The expected length of the sequence"""
+
+
+class SegmentPlan(SequencePlan):
+    """Metadata and expected properties for an included segment."""
+
+    model_config = ConfigDict(use_enum_values=True)
+
+    name: SegmentName
     """The name of the segment"""
 
-    required: bool
+    required: SegmentRule
     """Whether this segment must be present in all additions."""
 
-    length: int | None = None
+
+class MonopartitePlan(SequencePlan):
+    """Expected properties for an acceptable monopartite isolate."""
 
 
-class OTUSchema(BaseModel):
+class MultipartitePlan(BaseModel):
+    """Expected segments for an acceptable multipartite isolate."""
+
+    id: UUID4
+    """The unique id number of the multipartite plan"""
+
+    segments: list[SegmentPlan]
+
+    @computed_field
+    def required_segments(self) -> list[SegmentPlan]:
+        """Return a list of segments that are required by all additions."""
+        return [segment for segment in self.segments if segment.required]
+
+
+class IsolatePlan(BaseModel):
     """A schema for the intended data."""
 
     molecule: Molecule
     """The molecular metadata for this OTU."""
 
-    segments: list[Segment]
-    """The segments contained in this OTU."""
+    parameters: MonopartitePlan | MultipartitePlan
+    """The expected parameters of an acceptable isolate."""
 
     @computed_field
     def multipartite(self) -> bool:
         """Is true if multiple sequences are required to form a complete isolate."""
-        return len(self.segments) > 1
+        return type(self.parameters) is MultipartitePlan
+
+    @property
+    def segments(self) -> list[SegmentPlan | MonopartitePlan]:
+        """The segments contained in this OTU.
+
+        This property is a stopgap and will be removed in a future release.
+        """
+        if self.multipartite:
+            return self.parameters.segments
+
+        return [self.parameters]
```

Comment:
This can be implemented on `MonopartitePlan`. Not sure what we want to do when `name` is `None`.